_integration-tests/utils/generator/main.go

```diff
@@ -57,83 +70,124 @@ func main() {
 		if !pkg.IsDir() || pkg.Name() == "testdata" {
 			continue
 		}
-		testDir := path.Join(root, pkg.Name())
-		out := path.Join(testDir, genTestName)
-		pkgName, testCases := parseCode(testDir)
-		f := testFile(pkgName)
-
-		for _, tc := range testCases {
-			testSuffix := ""
-			if n := strings.TrimPrefix(tc, "TestCase"); n != "" {
-				testSuffix = "_" + n
-			}
 
-			f.
-				Func().
-				Id("TestIntegration_" + pkgName + testSuffix).
-				Params(jen.Id("t").Op("*").Qual("testing", "T")).
-				Block(
-					// utils.RunTest(t, new(TestCase))
-					jen.Qual(utilsPkg, "RunTest").Call(jen.Id("t"), jen.New(jen.Id(tc))),
-				).
-				Line().Empty()
+		testDir := path.Join(root, pkg.Name())
+		testData := parseCode(testDir)
+		if len(testData.Cases) == 0 {
+			continue
```

Comment:
Why not return an error there ? There is definitively an error if a test package has no tests
```suggestion
			log.Fatalf("Test package %q has no tests cases", pkg.Name())
```

Reply:
There is one directory that contains no test (I guess I should move that around instead of putting it under the carpet)